src/geometries/primitives/parametrizedcurve.jl

```diff
@@ -0,0 +1,41 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParametrizedCurve(a, b, func)
+
+A parametrized curve is a curve defined by a function `func` that maps a parameter `t` to a `Point` in space.
+The parameter `t` is defined in the interval `[a, b]`.
+
+## Examples
+
+```julia
+ParametrizedCurve(0.0, 1.0, t -> Point(cospi(2t), sinpi(2t)))
```

Comment:
Do you envision other use cases that involve non-Cartesian coordinates or this primitive is mostly for parametrizations of the form t -> x(t), y(t), z(t)?

Regarding the range of t, do you plan use cases with intervals other than [0,1]? Classically, a parametrized curve alpha is a function alpha: [0,1] --> R^n. If it is differentiable, it is often useful to reparametrize the curve by the arc length. See Differential Geometry of Curves and Surfaces by Manfredo (https://www.amazon.com/Differential-Geometry-Curves-Surfaces-Mathematics/dp/0486806995).

I wonder what would be the most ergonomic definition of constructors for the applications you have in mind. Maybe users would prefer to simply define the functions x(t), y(t), ... directly instead of manually handing the `Point` and CRS types?

```julia
x(t) = ...
y(t) = ...

c = ParametrizedCurve(x, y)
```